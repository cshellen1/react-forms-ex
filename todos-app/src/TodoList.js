import React, { useState } from 'react'
import { v4 as uuidv4 } from 'uuid';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

function TodoList() {
  const [todos, setTodos] = useState([]);

  const addTodo = (newTodo) => {
    setTodos(todos => [...todos, {
      ...newTodo,
      id: uuidv4()
    }])
  }

  const removeTodo = (todoId) => {
    setTodos(todos => todos.filter(todo => todo.id !== todoId))
  }
  
  return (
    <div>
      <h1>Todo List</h1>
      <NewTodoForm addTodo={addTodo} />
      <ul>
      {todos.map((todo) => (
        <Todo
          key={todo.id}
          id={todo.id}
          todo={todo.todo}
          removeTodo={removeTodo}
        />
      ))}
      </ul>
    </div>
  )
}

export default TodoList