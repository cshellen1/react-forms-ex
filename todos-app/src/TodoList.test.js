import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

function addTodo(getByLabelText, getByText ) {
  const todoInput = getByLabelText('Task');
  const btn = getByText('Add');
  
  fireEvent.change(todoInput, { target: { value: 'clean kitchen' } });
  fireEvent.click(btn);
}

it('renders without crashing', function () {
    render(<TodoList />);
});

it('matches snapshot', function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it('should add new todo', function () {
    const { getByLabelText, queryByText, getByText } = render(<TodoList />);
    // make sure there are no todos yet
    expect(queryByText('X')).not.toBeInTheDocument();
    addTodo(getByLabelText, getByText);
    // expect to see the button contained in the new box
    expect(getByText('X'));
    expect(getByText('clean kitchen'));
  });

  it('should remove a todo', function () {
    const { getByLabelText, queryByText, getByText } = render(<TodoList />);
    addTodo(getByLabelText, getByText);
  
    const removeBtn = getByText('X');
  
    fireEvent.click(removeBtn);
    expect(removeBtn).not.toBeInTheDocument();
  })
