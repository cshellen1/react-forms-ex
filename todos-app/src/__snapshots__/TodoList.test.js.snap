// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Todo List
    </h1>
    <form>
      <label
        for="todo"
      >
        Task
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="todo"
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
